### PR TITLE
Solver

### LINKED ISSUE #25883: Solving equations like Eq(Mod(192*i0, 2304), 0)
I'd like to be able to solve equations like

```py
Eq(Mod(192*i0, 2304), 0)
```

where `i0` is defined with `integer=True`. solve() currently cannot handle this equation. solveset can:

```py
>>> solveset(Eq(Mod(192*i0, 2304), 0), i0, Integers)
ImageSet(Lambda(_n, 12*_n), Integers)
```

But the problem is that the output of solveset isn't useful as is. 

Ideally, I'd like to get `Eq(Mod(i0, 12), 0)`. 

What would be the best way to implement this? Some alternatives I can think of are:

- Implement support in solve(). The solution `Eq(Mod(i0, 12), 0)` is a bit odd for solve though. I'm not sure how it would fit in the API.
- Add some sort of function to translate the set `{12⋅n │ n ∊ ℤ}` to a boolean.
- A new function for solving/simplifying modular equations.

I think the algorithms aren't the hard part here. For now I just want to be able to solve basic linear modular equations. I'm just unclear for now where this should go.

### PR DESCRIPTION
**Pull request by ParamThakkar123**
#### References to other Issues or PRs

Fixes #25883 

#### Brief description of what is fixed or changed
Added solve_modular function which can solve modular equations

#### Release Notes

* solvers
  * Added a new solver for modular equations.

### CODE REVIEW

**Comment by ParamThakkar123** [comment]
My username is not present in the .mailmap file. How can I add it to the mailmap ? The only check that fails is because my username is not added to the mailmap. I did it in my earlier pull request. But maybe it is not approved yet

**Comment by ParamThakkar123** [comment]
@oscarbenjamin please have a look at my pull request

**Comment by ParamThakkar123** [comment]
@sylee957 Please have a look at my pull request

File: sympy/printing/mathml.py [hunk 1]
```diff
@@ -428,6 +428,15 @@ def _print_Pow(self, e):
...
         return x
 
+    def _print_Inv_Trig(self, e):
+        if(self._settings['inv_trig_style']):
+            x = self.dom.createElement('apply')
+            inv_trig_symbol = self.mathml_tag(e)
+            x.appendChild(inv_trig_symbol)
+            arg_elem = self._print(e)
+            x.appendChild(arg_elem)
+            return x
```
Comment (oscarbenjamin) [inline sympy/printing/mathml.py:438]: These changes seem unrelated to the PR.

File: sympy/solvers/solvers.py [hunk 2]
```diff
@@ -1296,6 +1295,23 @@ def _has_piecewise(e):
...
+    domain (Set): The solution domain. Default is the set of real numbers.
+
+    Returns:
+    Set: The solution set.
+    """
+    if not isinstance(equation, Eq) or not isinstance(equation.lhs, Mod) or equation.rhs != 0:
+        raise ValueError("Invalid Modular Equation")
+
+    expression, modulus = equation.lhs.args
+    return solve(expression % modulus, symbol, domain=domain)
```
Comment (oscarbenjamin) [inline sympy/solvers/solvers.py:1314]: This is not what is intended by the linked issue.

**Comment by oscarbenjamin** [comment]
These changes are not needed and miss the point of the linked issue.
https://github.com/sympy/sympy/issues/25883

If the function here that just calls `solve(equation % modulus)` actually works
then the issue is already fixed. A new `solve_modular` function is not needed
just to call `solve` like this.
